crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -243,15 +342,25 @@ impl SettlementEncoder {
         let traded_tokens: HashSet<_> = self
             .trades()
             .iter()
-            .flat_map(|trade| {
-                [
-                    trade.order.order_creation.buy_token,
-                    trade.order.order_creation.sell_token,
+            .flat_map(|order_trade| {
+                vec![
+                    order_trade.trade.order.order_creation.buy_token,
+                    order_trade.trade.order.order_creation.sell_token,
                 ]
```

Comment:
Here and 10 lines below `vec!` isn't needed. Plain array `[...]` is preferred because it doesn't allocate.